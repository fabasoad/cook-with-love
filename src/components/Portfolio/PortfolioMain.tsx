/// <reference path='./PortfolioMain.d.ts' />
import * as React from 'react'
import { useTranslation } from 'react-i18next'

import PortfolioItem from './PortfolioItem'

import imgArkaDeutscheLieferadresse from '../../assets/images/portfolio/deutsche_lieferadresse_website.jpg'
import imgArkaInvebstor from '../../assets/images/portfolio/invebstor.jpg'
import imgArkaMedisport from '../../assets/images/portfolio/medisport_website.jpg'
import imgArkaPrintAndPost from '../../assets/images/portfolio/print_and_post_website.jpg'
import imgArkaWebsite from '../../assets/images/portfolio/arka_website.png'
import imgArkaWorkPlus from '../../assets/images/portfolio/work_plus.png'
import imgArkaZakupki from '../../assets/images/portfolio/zakupki_website.png'
import imgBitcoinComGames from '../../assets/images/portfolio/bitcoin.com_games.png'
import imgBitcoinComGamesAdmin from '../../assets/images/portfolio/bitcoin.com_games_admin.png'
import imgBitcoinComGamesAndroid from '../../assets/images/portfolio/bitcoin.com_android.png'
import imgBitcoinComMarkets from '../../assets/images/portfolio/bitcoin.com_markets.png'
import imgLohikaHpeAlm from '../../assets/images/portfolio/alm.jpg'
import imgLohikaHpeAlmRobot from '../../assets/images/portfolio/alm_robot.png'
import imgLohikaHpeLabManagement from '../../assets/images/portfolio/lab_management.jpg'
import imgLohikaEarnin from '../../assets/images/portfolio/earnin.png'
import imgLohikaPrysm from '../../assets/images/portfolio/prysm_iphone_app.png'
import imgSMA from '../../assets/images/portfolio/sma.png'

export default function PortfolioMain() {
  const { t } = useTranslation()
  return (
    <div id="portfolio">
      <div className="container">
        <div className="section-title text-center center">
          <h2>{t('cook-with-love-portfolio-title')}</h2>
          <hr />
        </div>
        <div className="row">
          <PortfolioItem
            url="https://markets.bitcoin.com/"
            name="Markets"
            title={t('cook-with-love-portfolio-markets-title')}
            subtitle={t('cook-with-love-portfolio-markets-subtitle')}
            img={imgBitcoinComMarkets}
          />
          <PortfolioItem
            url="https://games.bitcoin.com/"
            name="Games"
            title={t('cook-with-love-portfolio-casino-web-app-title')}
            subtitle={t('cook-with-love-portfolio-casino-web-app-subtitle')}
            img={imgBitcoinComGames}
          />
          <PortfolioItem
            url="https://games.bitcoin.com/"
            name="Games Admin"
            title={t('cook-with-love-portfolio-casino-admin-web-app-title')}
            subtitle={t('cook-with-love-portfolio-casino-admin-web-app-subtitle')}
            img={imgBitcoinComGamesAdmin}
          />
          <PortfolioItem
            url="https://games.bitcoin.com/android"
            name="Games Android"
            title={t('cook-with-love-portfolio-casino-android-app-title')}
            subtitle={t('cook-with-love-portfolio-casino-android-app-subtitle')}
            img={imgBitcoinComGamesAndroid}
          />
          <PortfolioItem
            url="https://www.earnin.com/"
            name="Earnin"
            title={t('cook-with-love-portfolio-earnin-title')}
            subtitle={t('cook-with-love-portfolio-earnin-subtitle')}
            img={imgLohikaEarnin}
          />
          <PortfolioItem
            url="https://www.prysm.com/software/iphone-app/"
            name="Prysm iPhone App"
            title={t('cook-with-love-portfolio-prysm-title')}
            subtitle={t('cook-with-love-portfolio-prysm-subtitle')}
            img={imgLohikaPrysm}
          />
          <PortfolioItem
            url="https://drive.google.com/drive/folders/163jhUuVrBCEYQ503sa1Y1M7KG--1gwaj?usp=sharing"
            name="Southern Marine Agency Website"
            title={t('cook-with-love-portfolio-sma-title')}
            subtitle={t('cook-with-love-portfolio-sma-subtitle')}
            img={imgSMA}
          />
          <PortfolioItem
            url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/Admin/sa_robot_toc.htm"
            name="ALM Robot"
            title={t('cook-with-love-portfolio-alm-robot-title')}
            subtitle={t('cook-with-love-portfolio-alm-robot-subtitle')}
            img={imgLohikaHpeAlmRobot}
          />
          <PortfolioItem
            url="https://www.microfocus.com/en-us/products/application-lifecycle-management/overview"
            name="ALM"
            title={t('cook-with-love-portfolio-alm-title')}
            subtitle={t('cook-with-love-portfolio-alm-subtitle')}
            img={imgLohikaHpeAlm}
          />
          <PortfolioItem
            url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/lm_ag.htm"
            name="Lab Management"
            title={t('cook-with-love-portfolio-alm-lab-management-title')}
            subtitle={t('cook-with-love-portfolio-alm-lab-management-subtitle')}
            img={imgLohikaHpeLabManagement}
          />
          <PortfolioItem
            url="https://youtu.be/8wTcyZEtQx8"
            name="Invebstor"
            title={t('cook-with-love-portfolio-invebstor-title')}
            subtitle={t('cook-with-love-portfolio-invebstor-subtitle')}
            img={imgArkaInvebstor}
          />
          <PortfolioItem
            url="https://youtu.be/eRxTKZ0knsg"
            name="Work+"
            title={t('cook-with-love-portfolio-work-plus-title')}
            subtitle={t('cook-with-love-portfolio-work-plus-subtitle')}
            img={imgArkaWorkPlus}
          />
          <PortfolioItem
            url="https://www.arka-media.de/"
            name="ARKA Website"
            title={t('cook-with-love-portfolio-arka-title')}
            subtitle={t('cook-with-love-portfolio-arka-subtitle')}
            img={imgArkaWebsite}
          />
          <PortfolioItem
            url="https://www.deutsche-lieferadresse.com/"
            name="Deutsche Lieferadresse Website"
            title={t('cook-with-love-portfolio-deutsche-lieferadresse-title')}
            subtitle={t('cook-with-love-portfolio-deutsche-lieferadresse-subtitle')}
            img={imgArkaDeutscheLieferadresse}
          />
          <PortfolioItem
            url="http://medisport-garmisch.de/"
            name="Medisport Website"
            title={t('cook-with-love-portfolio-medisport-title')}
            subtitle={t('cook-with-love-portfolio-medisport-subtitle')}
            img={imgArkaMedisport}
          />
          <PortfolioItem
            url="#portfolio"
            name="Print &amp; Post Website"
            title={t('cook-with-love-portfolio-print-and-post-title')}
            subtitle={t('cook-with-love-portfolio-print-and-post-subtitle')}
            img={imgArkaPrintAndPost}
          />
          <PortfolioItem
            url="https://zakupki-de.com.ua/"
            name="Zakupki Website"
            title={t('cook-with-love-portfolio-zakupki-title')}
            subtitle={t('cook-with-love-portfolio-zakupki-subtitle')}
            img={imgArkaZakupki}
          />
        </div>
      </div>
    </div>
  )
}
