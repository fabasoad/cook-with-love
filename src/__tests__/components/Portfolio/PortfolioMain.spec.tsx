/// <reference types="jest" />
import * as React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import PortfolioMain from '../../../components/Portfolio/PortfolioMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeAll(() => {
  tMock = useTranslation().t
})

test('should render PortfolioMain correctly', () => {
  const wrapper: ShallowWrapper = shallow(<PortfolioMain />)
  expect(wrapper).toMatchSnapshot()
  expect(tMock).toHaveBeenCalledTimes(35)
  expect(tMock).toHaveBeenNthCalledWith(1, 'cook-with-love-portfolio-title')
  for (let i = 1; i < 34; i += 2) {
    expect(tMock).toHaveBeenNthCalledWith(
      i + 1, expect.stringMatching(/^cook-with-love-portfolio-.*-title$/)
    )
    expect(tMock).toHaveBeenNthCalledWith(
      i + 2, expect.stringMatching(/^cook-with-love-portfolio-.*-subtitle$/)
    )
  }
})
