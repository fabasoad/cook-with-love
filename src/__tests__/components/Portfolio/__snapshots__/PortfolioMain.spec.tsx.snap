// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PortfolioMain correctly 1`] = `
<div
  id="portfolio"
>
  <div
    className="container"
  >
    <div
      className="section-title text-center center"
    >
      <h2>
        cook-with-love-portfolio-title
      </h2>
      <hr />
    </div>
    <div
      className="row"
    >
      <PortfolioItem
        img="test-file-stub"
        name="Markets"
        subtitle="cook-with-love-portfolio-markets-subtitle"
        title="cook-with-love-portfolio-markets-title"
        url="https://markets.bitcoin.com/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Games"
        subtitle="cook-with-love-portfolio-casino-web-app-subtitle"
        title="cook-with-love-portfolio-casino-web-app-title"
        url="https://games.bitcoin.com/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Games Admin"
        subtitle="cook-with-love-portfolio-casino-admin-web-app-subtitle"
        title="cook-with-love-portfolio-casino-admin-web-app-title"
        url="https://games.bitcoin.com/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Games Android"
        subtitle="cook-with-love-portfolio-casino-android-app-subtitle"
        title="cook-with-love-portfolio-casino-android-app-title"
        url="https://games.bitcoin.com/android"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Earnin"
        subtitle="cook-with-love-portfolio-earnin-subtitle"
        title="cook-with-love-portfolio-earnin-title"
        url="https://www.earnin.com/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Prysm iPhone App"
        subtitle="cook-with-love-portfolio-prysm-subtitle"
        title="cook-with-love-portfolio-prysm-title"
        url="https://www.prysm.com/software/iphone-app/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Southern Marine Agency Website"
        subtitle="cook-with-love-portfolio-sma-subtitle"
        title="cook-with-love-portfolio-sma-title"
        url="https://drive.google.com/drive/folders/163jhUuVrBCEYQ503sa1Y1M7KG--1gwaj?usp=sharing"
      />
      <PortfolioItem
        img="test-file-stub"
        name="ALM Robot"
        subtitle="cook-with-love-portfolio-alm-robot-subtitle"
        title="cook-with-love-portfolio-alm-robot-title"
        url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/Admin/sa_robot_toc.htm"
      />
      <PortfolioItem
        img="test-file-stub"
        name="ALM"
        subtitle="cook-with-love-portfolio-alm-subtitle"
        title="cook-with-love-portfolio-alm-title"
        url="https://www.microfocus.com/en-us/products/application-lifecycle-management/overview"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Lab Management"
        subtitle="cook-with-love-portfolio-alm-lab-management-subtitle"
        title="cook-with-love-portfolio-alm-lab-management-title"
        url="https://admhelp.microfocus.com/alm/en/12.60/online_help/Content/lm_ag.htm"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Invebstor"
        subtitle="cook-with-love-portfolio-invebstor-subtitle"
        title="cook-with-love-portfolio-invebstor-title"
        url="https://youtu.be/8wTcyZEtQx8"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Work+"
        subtitle="cook-with-love-portfolio-work-plus-subtitle"
        title="cook-with-love-portfolio-work-plus-title"
        url="https://youtu.be/eRxTKZ0knsg"
      />
      <PortfolioItem
        img="test-file-stub"
        name="ARKA Website"
        subtitle="cook-with-love-portfolio-arka-subtitle"
        title="cook-with-love-portfolio-arka-title"
        url="https://www.arka-media.de/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Deutsche Lieferadresse Website"
        subtitle="cook-with-love-portfolio-deutsche-lieferadresse-subtitle"
        title="cook-with-love-portfolio-deutsche-lieferadresse-title"
        url="https://www.deutsche-lieferadresse.com/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Medisport Website"
        subtitle="cook-with-love-portfolio-medisport-subtitle"
        title="cook-with-love-portfolio-medisport-title"
        url="http://medisport-garmisch.de/"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Print & Post Website"
        subtitle="cook-with-love-portfolio-print-and-post-subtitle"
        title="cook-with-love-portfolio-print-and-post-title"
        url="#portfolio"
      />
      <PortfolioItem
        img="test-file-stub"
        name="Zakupki Website"
        subtitle="cook-with-love-portfolio-zakupki-subtitle"
        title="cook-with-love-portfolio-zakupki-title"
        url="https://zakupki-de.com.ua/"
      />
    </div>
  </div>
</div>
`;
